src/node-saml/types.ts

```diff
@@ -3,12 +3,18 @@ import type { CacheProvider } from "./inmemory-cache-provider";
 export type SignatureAlgorithm = "sha1" | "sha256" | "sha512";
 
 export interface SamlSigningOptions {
-  privateKey?: string | Buffer;
+  privateKey: string | Buffer;
   signatureAlgorithm?: SignatureAlgorithm;
   xmlSignatureTransforms?: string[];
   digestAlgorithm?: string;
 }
 
+export const isValidSamlSigningOptions = (
```

Comment:
It might work too though that one is more specific. I have to admit, I like type guards and probably over use them, so I'm happy to try switching it to assertRequired if you feel that's better

Reply:
After having a closer look, it looks like assertRequired throws, and I don't think that's what we want here